std/net/curl.d

```diff
@@ -2449,6 +2449,9 @@ struct HTTP
 
                     auto m = header.findSplit(": ");
                     auto fieldName = m[0].toLower();
+                    ///Fixes https://issues.dlang.org/show_bug.cgi?id=24458
+                    if (fieldName is m[0])
+                        fieldName = fieldName.idup;
```

Comment:
This should probably become an unconditional `idup` when the invalid Associative Array key type conversion bug gets fixed

Reply:
Makes sense

Reply:
@dkorpel Fixed it

Reply:
That's a double copy in the worst (usual?) case. I guess that's fine; alternatively, it could be an `else fieldName = fieldName2.assumeUnique;` OSLT.

Reply:
What we need is a `toLower` that always copies.

Reply:
@CyberShadow Made it and put type hints to it becomes clearer the fieldName is immutable

Reply:
How about

```d
import std.exception : assumeUnique;

auto alwaysCopy(alias transformer, T)(T value)
{
    auto newValue = transformer(value);
    if (newValue is value)
        return value.idup;
    else
        return newValue.assumeUnique;
}

void main()
{
    import std.string : toLower;

    string s;
    s = "foo"; assert(s.alwaysCopy!toLower !is s);
    s = "Foo"; assert(s.alwaysCopy!toLower !is s);
}
```

Reply:
Would that go inside std.exception library then? Btw I don't really understand why assumeUnique is related to exception but anyway.
Please tell me if there is something else need to be done.

Reply:
I thought it was good to begin with. If the test gets stuck again, I'll merge manually.

Reply:
> Please tell me if there is something else need to be done.

We need to figure out what's with the required `dlang.phobos` check that's not letting this get green.